### PR TITLE
git-hack pushes new branches

### PR DESCRIPTION
**Pull request by kevgo**
@charlierudolph @allewun @ricmatsui 

* makes `git-hack` always push a new branch to the remote
* implements #664 
* prerequisite for #663

### CODE REVIEW

File: documentation/commands/git-hack.md [hunk 1]
```diff
@@ -13,9 +13,10 @@ git hack (--abort | --continue)
 
 #### DESCRIPTION
 
-Syncs the given parent branch,
+Syncs the given parent branch (default: main branch),
 forks a new feature branch with the given name off it,
-and brings over all uncommitted changes to the new feature branch.
+brings over all uncommitted changes to the new feature branch,
+and pushes everything to the remote repository.
```
Comment (charlierudolph) [inline documentation/commands/git-hack.md]: To me this makes it sound like it pushes up the previously uncommitted changes. The order here should probably be
```
Syncs the given parent branch (default: main branch),
forks a new feature branch with the given name off it,
pushes the new feature branch to the remote repository,
and brings over all uncommitted changes to the new feature branch.
```

Comment (kevgo) [inline documentation/commands/git-hack.md]: I like it.

**Comment by kevgo** [comment]
All comments addressed

File: documentation/commands/git-hack.md [hunk 2]
```diff
@@ -13,9 +13,10 @@ git hack (--abort | --continue)
 
 #### DESCRIPTION
 
-Syncs the given parent branch,
+Syncs the given parent branch (default: main branch),
 forks a new feature branch with the given name off it,
-and brings over all uncommitted changes to the new feature branch.
+pushes the new feature branch to the remote repository,
+and brings over all uncommitted changes to the new feature branch,
```
Comment (charlierudolph) [inline documentation/commands/git-hack.md]: `.` instead of  `,`

**Comment by charlierudolph** [comment]
Changes look fine to me. I like a bit more agreement on #664 before we actually merge this though

**Comment by kevgo** [comment]
I think we agreed that this is fine, and to add a config option to not push separately from this PR. So this should be good to go once #666 is shipped.

**Comment by charlierudolph** [comment]
:ship:

**Pull request closed without merging**